src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.cs

```diff
@@ -25,5 +28,46 @@ internal static partial class Crypto
         // returns the maximum length of a DER encoded ECDSA signature created with this key.
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcDsaSize")]
         internal static extern int EcDsaSize(SafeEcKeyHandle ecKey);
+
+        // todo: move this to common cng and openssl
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_LookupNidByOid")]
+        private static extern int CryptoNative_LookupNidByOid(string oidValue, ref int nid);
+        internal static int OidToNid(Oid oid)
+        {
+            if (oid == null)
+            {
+                Debug.Assert(false);
+                throw new ArgumentNullException("oid");
+            }
+
+            if (string.IsNullOrEmpty(oid.Value) && string.IsNullOrEmpty(oid.FriendlyName))
+            {
+                Debug.Assert(false);
+                throw new ArgumentException("oid");
```

Comment:
The first parameter to ArgumentException's ctor is a message, not the parameter name.  Unfortunate mistake in API design.